url.bs

```diff
@@ -88,6 +88,334 @@ valid input. User agents, especially conformance checkers, are encouraged to rep
+   <td>
+    <p>The URL has a <a>special scheme</a> and it uses U+005C (\) instead of U+002F (/).
+    <p class=example id=example-unexpected-reverse-solidus>"<code>https://example.org\path\to\file</code>"
+   <td>❌
+  <tr>
+   <td><dfn>missing-solidus-before-authority</dfn>
+   <td>
+    <p>The URL <a>includes credentials</a> that are not preceded by "<code>//</code>".
+    <p class=example id=example-missing-solidus-before-authority>"<code>https:user@example.org</code>"
+   <td>❌
+  <tr>
+   <td><dfn>unexpected-at-sign</dfn>
+   <td>
+    <p>The URL <a>includes credentials</a>, however this is considered invalid.
+    <p class=example id=example-unexpected-at-sign>"<code>https://user@example.org</code>"
+   <td>❌
+  <tr>
+   <td><dfn>unexpected-credentials-without-host</dfn>
+   <td>
+    <p>The URL <a>include credentials</a>, but no <a for=url>host</a>.
+    <p class=example id=example-unexpected-credentials-without-host>"<code>https://user:pass@</code>"
+   <td>✅
+  <tr>
+   <td><dfn>unexpected-port-without-host</dfn>
+   <td>
+    <p>The URL contains a <a for=url>port</a>, but no <a for=url>host</a>.
+    <p class=example id=example-unexpected-port-without-host>"<code>https://:443</code>"
+   <td>✅
+  <tr>
+   <td><dfn>empty-host-special-scheme</dfn>
+   <td>
+    <p>The URL has a <a>special scheme</a>, but does not contain a <a for=url>host</a>.
+    <p class=example id=example-empty-host-special-scheme>"<code>https://#fragment</code>"
+   <td>✅
+  <tr>
+   <td><dfn>host-invalid</dfn>
+   <td>
+    <p>The <a for=url>host</a> portion of the URL is an empty string when it
+    <a>includes credentials</a> or a <a>port</a> and the <a>basic URL parser</a>'s state is
+    overridden.
+    <pre class=example id=example-host-invalid><code class=lang-javascript>
+const url = new URL("https://example:9000");
+url.hostname = "";</code></pre>
+   <td>❌
+  <tr>
+   <td><dfn>port-out-of-range</dfn>
+   <td>
+    <p>The input's port is too big.
+    <p class=example id=example-port-out-of-range>"<code>https://example.org:70000</code>"
+   <td>✅
+  <tr>
+   <td><dfn>port-invalid</dfn>
+   <td>
+    <p>The input's port is invalid.
+    <p class=example id=example-port-invalid>"<code>https://example.org:7z</code>"
+   <td>✅
+  <tr>
+   <td><dfn>unexpected-Windows-drive-letter</dfn>
+   <td>
+    <p>The input is a <a>relative-URL string</a> that <a>starts with a Windows drive letter</a> and
+    the <a>base URL</a>'s <a for=url>scheme</a> is "<code>file</code>".
+    <pre class=example id=example-unexpected-windows-drive-letter><code class=lang-javascript>
+let url = new URL("/c:/path/to/file", "file:///c:/");</code></pre>
+   <td>❌
+  <tr>
+   <td><dfn>unexpected-Windows-drive-letter-host</dfn>
+   <td>
+    <p>The file URL's host is a Windows drive letter.
+    <p class=example id=example-unexpected-windows-drive-letter-host>"<code>file://c:</code>"
+   <td>❌
+ <tbody>
+  <tr>
+   <th colspan=3 scope=rowgroup>URL parsing and <a>opaque-host parser</a>
```

Comment:
I don't really understand why this is split out from the above section.

Reply:
I wanted to group validation errors per somewhat major section. However these validation errors were reused across Hosts and URLs. I could just put it under Hosts as that's the first appearance, but it might look surprising?

Reply:
Yeah, I think just putting them under URLs would make more sense. Unlike IPv4/IPv6/ToASCII, there's no clear differentiation here IMO.